spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/message/EntryEventMessagePayload.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.hazelcast.message;
+
+import org.springframework.util.Assert;
+
+/**
+ * Hazelcast Message Payload for Entry Events
+ *
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ */
+public class EntryEventMessagePayload<K, V> {
```

Comment:
I understand your position to provide some clear way for end-user to get deal with Hazelcast Map changes, but we should be sure that we don't provide some `ClassLoader` issue for distributed SI application.
For me this looks like the raw `Map`, which can be simply converted to any other downstream system type, e.g. JSON for HTTP or AMQP request.

Nevertheless I see the simple way to overcome that with on short component using SpEL `inline Map` support:
````xml
<transformer expression="{key: payload.key, value: payload.value, oldValue: payload.oldValue, 
                                    numberOfEntriesAffected: payload.numberOfEntriesAffected}"/>
````
At least we should mention this in the Manual.

CC: @garyrussell